### ISSUE TITLE
react/prop-types: destructured component props with 'stringed-keys:'  triggers "missing in props validation"

### ISSUE THREAD

**Issue by faceyspacey** [gajus/eslint-plugin-flowtype#188]
For example:

```
type Props = {
  'data-hover': string
}

function MyComponent({
  'data-hover': dataHover
}: Props) {
   return <span data-hover={dataHover} />
}
```

will return this error message:
```
file: '/MyComponent.js'
severity: 'Error'
message: ''data-hover' is missing in props validation (react/prop-types)'
at: '6,3'
source: 'eslint'
```

##### VS Code + Atom

**Comment by danharper** [comment]
That error's coming from the React ESLint plugin's [`react/prop-types`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md) rule.

Looks like it enforces you declare propTypes too:

```js
type Props = {
  'data-hover': string
}

function MyComponent({
  'data-hover': dataHover
}: Props) {
   return <span data-hover={dataHover} />
}

MyComponent.propTypes = {
  'data-hover': PropTypes.string.isRequired,
}
```

**Comment by faceyspacey** [comment]
but you don't have to do that with regular keys. Also, isn't the whole point that if you're using flow and eslint correctly (and your project isn't a library project used in other projects) that you don't have to do runtime checking anymore with `propTypes`?

**Comment by danharper** [comment]
Yeah, but it's not a bug in this project - it's from `eslint-plugin-react`.

If you don't need PropTypes, I'd just disable that rule: `"react/prop-types": 0`

**Comment by faceyspacey** [comment]
I'll look into that other project, but if you disable it you lose the benefits of eslint telling you that you don't have static prop types defined.
